lib/ansible/module_utils/network/iosxr/iosxr.py

```diff
@@ -63,87 +102,282 @@ def check_args(module, warnings):
+
+                if meta.get('attrib', None) and opcode in ('delete', 'merge'):
+                    child.set(BASE_1_0 + meta.get('attrib'), opcode)
+
+                continue
+
+            text = None
+            param_key = key.split(":")
+            if param_key[0] == 'a':
+                if param.get(param_key[1], None):
+                    text = param.get(param_key[1])
+            elif param_key[0] == 'm':
+                if meta.get('value', None):
+                    text = meta.get('value')
+
+            if text:
+                if meta.get('ns', None) is True:
+                    child = etree.SubElement(parent, candidates[-1], nsmap=NS_DICT[key.upper() + "_NSMAP"])
+                else:
+                    child = etree.SubElement(parent, candidates[-1])
+                child.text = text
 
+    if len(meta_subtree) > 1:
+        for item in meta_subtree:
+            container_ele.append(item)
 
-def load_config(module, commands, warnings, commit=False, replace=False, comment=None, admin=False):
-    cmd = 'configure terminal'
-    if admin:
-        cmd = 'admin ' + cmd
+    return sub_root
 
-    rc, out, err = exec_command(module, cmd)
-    if rc != 0:
-        module.fail_json(msg='unable to enter configuration mode', err=to_text(err, errors='surrogate_or_strict'))
 
-    failed = False
-    for command in to_list(commands):
-        if command == 'end':
-            continue
+def build_xml(container, xmap=None, params=None, opcode=None):
+    if not HAS_XML:
+        return 'no_lxml'
 
-        rc, out, err = exec_command(module, command)
-        if rc != 0:
-            failed = True
-            break
+    if opcode == 'filter':
+        root = etree.Element("filter", type="subtree")
+    elif opcode in ('delete', 'merge'):
+        root = etree.Element("config", nsmap=NS_DICT['BASE_NSMAP'])
 
-    if failed:
-        exec_command(module, 'abort')
-        module.fail_json(msg=to_text(err, errors='surrogate_or_strict'), commands=commands, rc=rc)
+    container_ele = etree.SubElement(root, container, nsmap=NS_DICT[container.upper() + "_NSMAP"])
 
-    rc, diff, err = exec_command(module, 'show commit changes diff')
-    if rc != 0:
-        # If we failed, maybe we are in an old version so
-        # we run show configuration instead
-        rc, diff, err = exec_command(module, 'show configuration')
+    if xmap:
+        if not params:
+            build_xml_subtree(container_ele, xmap)
+        else:
+            subtree_list = list()
+
+            for param in to_list(params):
+                subtree_list.append(build_xml_subtree(container_ele, xmap, param, opcode=opcode))
+
+            for item in subtree_list:
+                container_ele.append(item)
+
+    return root
+
+
+def get_config_diff(module, running=None, candidate=None):
+    conn = get_connection(module)
+    capabilities = get_device_capabilities(module)
+    network_api = capabilities.get('network_api')
```

Comment:
SInce network_api check is used in multiple lines can we move it out in another function?

Reply:
This may move out a couple of lines only.